clang/test/SemaSYCL/reqd-work-group-size-device-direct-prop.cpp

```diff
@@ -0,0 +1,81 @@
+// RUN: %clang_cc1 -fsycl-is-device -internal-isystem %S/Inputs -fsyntax-only -sycl-std=2020 -verify -DTRIGGER_ERROR %s
+// RUN: %clang_cc1 -fsycl-is-device -internal-isystem %S/Inputs -sycl-std=2020 -ast-dump %s | FileCheck %s
+
+// Test for AST of reqd_work_group_size kernel attribute in SYCL 2020.
+
+#include "sycl.hpp"
+
+using namespace cl::sycl;
+queue q;
+
+// The GNU and [[cl::reqd_work_group_size]] spellings are deprecated in SYCL
+// mode, and still requires all three arguments.
+__attribute__((reqd_work_group_size(4, 4, 4))) void four_once_more(); // expected-warning {{attribute 'reqd_work_group_size' is deprecated}} \
+                                                                      // expected-note {{did you mean to use '[[sycl::reqd_work_group_size]]' instead?}}
+[[cl::reqd_work_group_size(4, 4, 4)]] void four_with_feeling(); // expected-warning {{attribute 'cl::reqd_work_group_size' is deprecated}} \
+                                                                // expected-note {{did you mean to use 'sycl::reqd_work_group_size' instead?}}
+
+#ifdef TRIGGER_ERROR
+__attribute__((reqd_work_group_size(4))) void four_yet_again(); // expected-error {{'reqd_work_group_size' attribute requires exactly 3 arguments}} \
+                                                                // expected-warning {{attribute 'reqd_work_group_size' is deprecated}} \
+                                                                // expected-note {{did you mean to use '[[sycl::reqd_work_group_size]]' instead?}}
+
+[[cl::reqd_work_group_size(4)]] void four_with_more_feeling(); // expected-error {{'reqd_work_group_size' attribute requires exactly 3 arguments}} \
+                                                               // expected-warning {{attribute 'cl::reqd_work_group_size' is deprecated}} \
+                                                               // expected-note {{did you mean to use 'sycl::reqd_work_group_size' instead?}}
+#endif // TRIGGER_ERROR
```

Comment:
Why this `#endif` and then `#ifdef TRIGGER_ERROR` again?

Reply:
my mistake . Fixed now